main.js

```diff
@@ -0,0 +1,85 @@
+const inViewport = require('in-viewport');
+
+//find all convo cards
+const convoCards = document.getElementsByClassName('conversation');
+
+//for each card, define inViewport callback function
+[].slice.call(convoCards).forEach(card => inViewport(card, { offset: -200 }, function() {
+
+    //get card's classes so it can be used as a querying string
+    const cardSelector = card.className.split(" ").join(".");
+
+    //find all chat bubbles inside card
+    const chatList = document.querySelectorAll(`.${cardSelector} .bubble`);
+
+    //for each bubble, apply animation delay as indicated by element's data-delay attribute
+    [].slice.call(chatList).forEach(text => {
```

Comment:
very helpful, thanks! useful for me to know how I can make my code more readable for others